### PR TITLE
Check for dependencies when installing

### LINKED ISSUE #43: Installer should offer to install relevant dependencies
The installer should install the relevant OpenTelemetry dependencies for your project and suggest relevant next steps after installation.

For example, if your `requirements.txt` contains `django`, then we should offer to add `opentelemetry-instrumentation-django` to the `requirements.txt` for you. This is important because the dependency being installed is what will cause AppSignal to start that instrumentation. The same could be done for all other instrumentations.

The installer should also give you potential "next steps" instructions based on those dependencies. For example, if it installed `opentelemetry-instrumentation-django` for you, then it should suggest that you add the code snippet that initialises AppSignal to your `manage.py` file. For many instrumentations, there will be no obvious "next step" that is more specific than the generic "add this to your application's entrypoint" -- that's okay.

I don't think we need to run `pip install` for users after we add the new dependencies, but we could. I don't think we need to write the initialise code to `manage.py` ourselves, either.

To do
---

- [x] Detect relevant existing dependencies from `requirements.txt`
- [x] Write relevant OpenTelemetry dependencies to `requirements.txt`
- [x] Suggest relevant next steps based on dependencies: install/config steps or link to docs

This is a wishlist -- if any part of this is hard, consider dropping it. For example, maybe we can just tell users which dependencies they need to add instead of writing them to the file ourselves.

### PR DESCRIPTION
**Pull request by luismiramirez**
The installation command checks for the presence of supported web frameworks to add the OpenTelemetry instrumentation dependencies if needed and to display a custom output with brief instructions about the detected framework if it requires it.

Closes #43 

[skip changeset]

### CODE REVIEW

**Review by unflxw — APPROVED** [review]
Some other thoughts:

- The Node.js 2.x installer would [ask you for permission to install the instrumentation-specific packages](https://github.com/appsignal/appsignal-javascript/blob/39438e9f0649177fe55243054e8b0e8de1bd4515/packages/cli/src/installers/node/index.ts#L78-L99) before messing with your `package.json`. I don't think this is a requirement, but I'm pointing it out in case this is something we want to do.
- The other supported instrumentations (`celery`, `jinja2`, `psycopg2`, `redis`, `requests`) could also be installed automatically. None of those need specific installation instructions, [except for Celery](https://github.com/appsignal/appsignal-docs/pull/1297/files#diff-3c5a887ca3621d8a47625e157c6948748b78af35b3f3930b541c76e2da200502L24-L27), which does have some ceremony to it.

File: src/appsignal/cli.py [hunk 1]
```diff
@@ -152,6 +148,74 @@ def _write_file(self):
...
+
+    def _generic_installation(self):
+        print("✅ Done! AppSignal for Python has now been installed.")
+        print()
+        print("To start AppSignal in your application, add the following code to your")
+        print("application's entrypoint:")
+        print()
+        print("    from __appsignal__ import appsignal")
+        print("    appsignal.start()")
+        print()
```
Comment (unflxw) [inline src/appsignal/cli.py:211]: This branch of the installation could also use a documentation link. Maybe to the list of instrumentations?

File: src/appsignal/cli.py [hunk 2]
```diff
@@ -152,6 +148,74 @@ def _write_file(self):
...
+
+    def _add_dependency(self, dependency_name):
+        requirement_file = self._requirements_file()
```
[the remark this answers is not in the document]
Comment (unflxw) [reply src/appsignal/cli.py:213]: Ah, my bad! I missed the `if not self._search_dependency("opentelemetry-...")` before `_add_dependency` is called. That one's on me.

File: src/appsignal/cli.py [hunk 3]
```diff
@@ -152,6 +148,74 @@ def _write_file(self):
...
+        if requirement_file:
+            with open(requirement_file, "a") as f:
+                f.write(f"{dependency_name}\n")
```
[the remark this answers is not in the document]
Comment (luismiramirez) [reply src/appsignal/cli.py:216]: Will 

```python
f.write(f\n"{dependency_name}\n")
```

help? I don't have a proper way to test it.

**Review by tombruijn — APPROVED** [review]
Looks like a good start. We should add the other supported packages to this flow too.

**Pull request merged**